fedimint-client/src/module/gen.rs

```diff
@@ -71,7 +76,12 @@ pub trait IClientModuleGen: IDynCommonModuleGen + Debug + MaybeSend + MaybeSync
 
     fn as_common(&self) -> &(dyn IDynCommonModuleGen + Send + Sync + 'static);
 
-    async fn init(&self, cfg: ClientModuleConfig, db: Database) -> anyhow::Result<DynClientModule>;
+    async fn init(
+        &self,
+        cfg: ClientModuleConfig,
+        db: Database,
+        instance_id: ModuleInstanceId,
```

Comment:
Not a blocker, but would be best if modules didn't directly know their id.

Reply:
This is a bit more involved, I was thinking about it but decided against it for now, will add a fixme.